### DISCUSSION TITLE
how to log metrics with None value at beginning at on_train_start

### DISCUSSION THREAD

**Discussion by quancs** [Lightning-AI/pytorch-lightning#9036] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
In my case, I don't know the initial value of `val/neg_si_sdr` and `val/fploss`. So, how can I set them to None at beginning?

Below is my code to set them to None at beginning:
```
    def on_train_start(self):
        self.logger.log_hyperparams(self.hparams, {"val/neg_si_sdr": None, "val/fploss": None})
```
But I receive this:
```
Exception: ValueError

 you tried to log None which is not currently supported. Try a dict or a scalar/tensor.

The above exception was the direct cause of the following exception:

  File "/mnt/home/quancs/projects/NBSS_pmt/model_NBSS_ifp.py", line 91, in on_train_start
    self.logger.log_hyperparams(self.hparams, {"val/neg_si_sdr": None, "val/fploss": None})
  File "/mnt/home/quancs/projects/NBSS_pmt/train.py", line 94, in main
    trainer.fit(model, datamodule=datamodule)
  File "/mnt/home/quancs/projects/NBSS_pmt/train.py", line 129, in <module>
    main(args)
```

**Comment by talhaanwarch** [comment]
You can log 0 instead of None.

**Reply by quancs** [reply]
For metric like mse, 0 is the best, we could not give it the best at the beginning.

**Reply by quancs** [reply]
Besides, when we set it to the best at the beginning, at the HPARAMS page of tensorboard, we could not compare which hyperparamter set is the best for out model, because all of them are the best.
